File: 06_java/src/DFS.java
import java.util.*;

public class DFS {
  public static boolean[] visited = new boolean[9];
  static int[][] graph = {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};

  public static void dfs(int x) {

    visited[x] = true;
    System.out.print(x + " ");
//    System.out.println("graph[x]: " + Arrays.toString(graph[x]));
    for (int node : graph[x]) {
//      System.out.println("node: " + node);
      if (!visited[node]) {
        dfs(node);
      }
    }
  }

  public static void main(String[] args) {
    dfs(1);
  }
}

File: 06_C/DFS.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int visited[9];
int graph[9][3] = {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};

void dfs(int x){
    visited[x] = 1;
    printf("%d ", x);
    int n = sizeof(graph[x]) / sizeof(int);
    for (int i = 0; i < n; i++) {
        int node = graph[x][i];
        if (!visited[node]) {
         dfs(node);
        }
    }
}

int main(void) {
    dfs(1); 
    return 0;
}